src/raw/http_util/resolver.rs

```diff
@@ -0,0 +1,69 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use hyper::client::connect::dns::Name;
+use once_cell::sync::OnceCell;
+use reqwest::dns::{Addrs, Resolve};
+use trust_dns_resolver::TokioAsyncResolver;
+
+/// a global resolver to reuse dns cache
+///
+/// # Note:
+/// this structure contains nothing, only used as an entry point
+/// to call from `trust_dns_resolver`
+#[derive(Debug)]
+pub struct DnsClient {}
+
+static GLOBAL_DNS_CLIENT: OnceCell<Arc<TokioAsyncResolver>> = OnceCell::new();
+
+/// Get a new async dns client.
+fn get_client() -> Arc<TokioAsyncResolver> {
+    GLOBAL_DNS_CLIENT
+        .get_or_init(|| {
+            let resolver = TokioAsyncResolver::tokio_from_system_conf().unwrap();
+            Arc::new(resolver)
+        })
+        .clone()
+}
+
+impl DnsClient {
```

Comment:
How about naming `DnsClient` to `TrustDnsResolver`? I plan to add `StdResolver` and `BlockingStdResolver` in the future.